### ISSUE TITLE
Cannot read property 'links' of undefined

### ISSUE THREAD

**Issue by CodeOfficer** [emberjs/data#1247]
I have a model as such ..

``` javascript
App.Event = DS.Model.extend({
  map: DS.belongsTo('map', {async: true})
});
```

When I fetch the model's map association with ...

``` javascript
model.get('map');
```

I see the following ....

> Cannot read property 'links' of undefined 
> (anonymous function) (vendor.js:53078)
> (anonymous function) (vendor.js:55707)
> (anonymous function) (vendor.js:17256)
> OrderedSet.forEach (vendor.js:17099)
> Map.forEach (vendor.js:17254)
> DS.Model.reopenClass.eachRelationship (vendor.js:55706)
> normalizeRelationships (vendor.js:53075)
> DS.Store.Ember.Object.extend.push (vendor.js:52868)
> (anonymous function) (vendor.js:53177)
> invokeCallback (vendor.js:22184)
> (anonymous function) (vendor.js:22234)
> EventTarget.trigger (vendor.js:22007)
> (anonymous function) (vendor.js:22301)
> DeferredActionQueues.flush (vendor.js:19543)
> Backburner.end (vendor.js:19629)
> Backburner.run (vendor.js:19668)
> Ember.run (vendor.js:19999)
> DS.RESTAdapter.DS.Adapter.extend.ajax.Ember.RSVP.Promise.hash.success (vendor.js:57514)
> jQuery.Callbacks.fire (vendor.js:3603)
> jQuery.Callbacks.self.fireWith (vendor.js:3715)
> done (vendor.js:8069)
> jQuery.ajaxTransport.send.callback (vendor.js:8491)

This appears to be because ....

https://github.com/emberjs/data/blob/master/packages/ember-data/lib/system/relationships/ext.js#L395

... is called without a 3rd 'data' parameter and so it's undefined when it hits this line ...

https://github.com/emberjs/data/blob/master/packages/ember-data/lib/system/store.js#L1284
